e2e-tests/tests/test_jolteon_two_chain_commit.py

```diff
@@ -0,0 +1,325 @@
+                    lock_advancements += 1
+                    logger.info(f"Lock advanced during consecutive QCs: {prev_state['r_lock']} -> {curr_state['r_lock']}")
+        
+        # Analyze lock advancement patterns
+        total_lock_advancements = 0
+        for i in range(1, len(consensus_history)):
+            if consensus_history[i]['r_lock'] > consensus_history[i-1]['r_lock']:
+                total_lock_advancements += 1
+        
+        logger.info(f"2-chain commit analysis:")
+        logger.info(f"  Consecutive QC pairs: {consecutive_qc_pairs}")
+        logger.info(f"  Lock advancements during consecutive QCs: {lock_advancements}")
+        logger.info(f"  Total lock advancements: {total_lock_advancements}")
+        
+        # Verify 2-chain commit behavior
+        if consecutive_qc_pairs > 0:
+            logger.info("✅ Consecutive QC pairs detected (2-chain pattern)")
+            
+            # Calculate commit rate during consecutive QCs
+            if consecutive_qc_pairs > 0:
+                commit_rate = lock_advancements / consecutive_qc_pairs
+                logger.info(f"Commit rate during consecutive QCs: {commit_rate:.2f}")
+                
+                # In Jolteon, we expect commits to happen when we have consecutive QCs
+                if commit_rate > 0:
+                    logger.info("✅ 2-chain commit rule appears to be working")
+                else:
+                    logger.warning("⚠️  No commits detected during consecutive QCs")
+        else:
+            logger.info("ℹ️  No consecutive QC pairs found - this may be normal for this implementation")
+        
+        # Verify basic safety properties
+        # Lock should never decrease
+        for i in range(1, len(consensus_history)):
+            assert consensus_history[i]['r_lock'] >= consensus_history[i-1]['r_lock'], \
+                f"Lock decreased: {consensus_history[i-1]['r_lock']} -> {consensus_history[i]['r_lock']}"
+        
+        # QC round should never decrease
+        for i in range(1, len(consensus_history)):
+            assert consensus_history[i]['qc_round'] >= consensus_history[i-1]['qc_round'], \
+                f"QC round decreased: {consensus_history[i-1]['qc_round']} -> {consensus_history[i]['qc_round']}"
+        
+        logger.info("✅ 2-chain commit rule test completed")
+
+    @mark.test_key('JOLTEON-2CHAIN-002')
+    def test_commit_latency_measurement(self, api: BlockchainApi, config: ApiConfig):
+        """Measure commit latency in Jolteon
+        
+        This test measures the time between QC formation and block commitment
+        to verify Jolteon's improved commit latency compared to 3-chain HotStuff.
+        """
+        logger.info("Starting Jolteon commit latency measurement test")
+        
+        # Monitor for QC formation and subsequent commits
+        wait_time = 120  # Monitor for 2 minutes
+        check_interval = 5  # Check every 5 seconds for higher resolution
```

Comment:
If you're looking for at least 1 state change then for sure you'll get it after a round timeout period (perhaps + a bit more time, not sure how we can quantify it though)